docs/user_guide/hdl_coding_guideline.rst

```diff
@@ -0,0 +1,960 @@
+.. _hdl_coding_guideline:
+
+ADI HDL coding guideline
+===============================================================================
+
+1. Introduction
+-------------------------------------------------------------------------------
+
+This document contains coding and documentation guidelines which must be
+followed by all HDL projects. The purpose of this document is to
+establish a set of rules that specify the layout, naming conventions and
+some general coding practices for the HDL modules implementation.
+Specific HDL coding practices meant to obtain maximum performance from a
+FPGA design are described by external documents such as
+`[1] <http://www.xilinx.com/support/documentation/white_papers/wp231.pdf>`__
+and
+`[2] <http://www.asic-world.com/code/verilog_tutorial/peter_chambers_10_commandments.pdf>`__
+and are not included in this document.
+
+There are two types of rules: *should* and *must* rules
+ * *Should* rules are advisory rules. They suggest the recommended way of doing things.
+ * **Must** rules are mandatory requirements.
+
+The coding rules are intended to be applied to HDL modules written using
+VHDL or Verilog.
+
+When an external IP is used, the naming conventions practiced by the IP
+*should* be kept, even if they do not match the rules specified in this
+document.
+
+2. Coding style
+-------------------------------------------------------------------------------
+
+A. Layout
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A1
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Spaces **must** be used instead of tabs.
+
+This allows the code to be properly visualized by any editor. **Do not**
+leave spaces at the end of a line. The following editor settings **must**
+be used: *Tab Size*: 2, *Indent Size*: 2.
+
+A2
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One white space **must** be inserted around operators, such as
+=, ==, &&, \|\|, &, \|, ^, etc.
+
+.. _example-a2:
+
+Example
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Incorrect:
+
+.. code-block::
+
+   if((my_signal1==1'b0)&&(my_bus[3:0]==4'd5))
+
+Correct:
+
+.. code-block::
+
+   if ((my_signal == 1'b0) && (my_bus[3:0] == 4'd5))
+
+A3
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The *always* block *should* have a space before \*\*@\*\* symbol.
+
+.. _example-a3:
+
+Example
```

Comment:
Can we remove the Example headers? It is ugly in the side panel

Reply:
Done